File: random/polymorphism.cc
#include <iostream>
using namespace std;

class Colour
{
public:
	Colour(){}
	virtual void say()
	{
		cout << "I'm a generic colour" << endl;
	}
};

class Purple : public Colour
{
public:
	Purple() : Colour() {}
	void say()
	{
		cout << "I'm purple!" << endl;
	}
};

class DarkPurple : public Purple
{
public:
	DarkPurple() {}
	void say()
	{
		cout << " I'm a DARK purple..." << endl;
	}
};

int main()
{
	Colour *a = new DarkPurple();
	a->say();
}

File: random/polymorphism.java
import java.lang.String;

public class polymorphism {
	public class Colour {
		public void say() {
			System.out.println("I'm a generic colour");
		}
	}

	public class Purple extends Colour {
		@Override
		public void say() {
			System.out.println("I'm purple");
		}
	}

	public static void main (String[] args) {
		polymorphism me = new polymorphism();
		Colour colour = me.new Purple();
		colour.say();
	}
}
